### PR TITLE
add pre-commit checks

### LINKED ISSUE #263: linting checks can be done via pre-commit
Maybe all these checks can be done via pre-commit as we started doing for other projects?
https://github.com/NSLS-II/scientific-python-cookiecutter/blob/main/%7B%7B%20cookiecutter.repo_name%20%7D%7D/.github/workflows/pre-commit.yml

_Originally posted by @mrakitin in https://github.com/bluesky/hklpy/pull/261#discussion_r1349014033_

### PR DESCRIPTION
**Pull request by prjemian**
- close #263

### CODE REVIEW

**Comment by prjemian** [comment]
> Since unit tests have been passing before this branch, comment out parts so that tests pass now as well.

In a separate issue, resolve issues [noted](https://github.com/bluesky/hklpy/issues/263#issuecomment-1766882681) when certain tests were commented out due to failure with existing code.

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  workflow_dispatch:
+
+jobs:
+  pre-commit:
+    # pull requests are a duplicate of a branch push if within the same repo.
+    if: github.event_name != 'pull_request' || github.event.pull_request.head.repo.full_name != github.repository
+
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
```
Comment (mrakitin) [inline .github/workflows/pre-commit.yml:17]: It can be v4 now.
```suggestion
      - uses: actions/checkout@v4
```

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+# -   repo: https://github.com/timothycrosley/isort
+#     rev: 5.12.0
+#     hooks:
+#     -   id: isort
+
+# -   repo: https://github.com/psf/black
+#     rev: 22.8.0
+#     hooks:
+#     -   id: black
```
Comment (mrakitin) [inline .pre-commit-config.yaml:37]: Did those commented-out checks fail?

**Review by mrakitin — COMMENTED** [review]
Looks good overall. Not sure why the linting checks were commented out.

**Comment by prjemian** [comment]
Yes, those checks failed as noted above. Figured that fixing then was not the job of this issue.

**Review by mrakitin — APPROVED** [review]

**Pull request merged**